code/modules/atmospherics/machinery/components/fusion/hypertorus.dm

```diff
@@ -0,0 +1,1376 @@
+	name = "HFR waste output port"
+	desc = "Waste port for the Hypertorus Fusion Reactor, designed to output the hot waste gases coming from the core of the machine."
+	icon_state = "waste_output_off"
+	icon_state_open = "waste_output_open"
+	icon_state_off = "waste_output_off"
+	icon_state_active = "waste_output_active"
+	circuit = /obj/item/circuitboard/machine/HFR_waste_output
+
+/obj/machinery/atmospherics/components/unary/hypertorus/moderator_input
+	name = "HFR moderator input port"
+	desc = "Moderator port for the Hypertorus Fusion Reactor, designed to move gases inside the machine to cool and control the flow of the reaction."
+	icon_state = "moderator_input_off"
+	icon_state_open = "moderator_input_open"
+	icon_state_off = "moderator_input_off"
+	icon_state_active = "moderator_input_active"
+	circuit = /obj/item/circuitboard/machine/HFR_moderator_input
+
+/obj/machinery/atmospherics/components/binary/hypertorus/core
+	name = "HFR core"
+	desc = "This is the Hypertorus Fusion Reactor core, an advanced piece of technology to finely tune the reaction inside of the machine. It has I/O for cooling gases."
+	icon = 'icons/obj/atmospherics/components/hypertorus.dmi'
+	icon_state = "core_off"
+	circuit = /obj/item/circuitboard/machine/HFR_core
+	pipe_flags = PIPING_ONE_PER_TURF | PIPING_DEFAULT_LAYER_ONLY
+	density = TRUE
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 50
+	var/icon_state_open = "core_open"
+	var/icon_state_off = "core_off"
+	var/icon_state_active = "core_active"
+	///Checks if the machine state is active (all parts are connected)
+	var/active = FALSE
+	///Checks if the user has started the machine
+	var/start_power = FALSE
+	///Checks for the cooling to start
+	var/start_cooling = FALSE
+	///Checks for the fuel to be injected
+	var/start_fuel = FALSE
+	///Checks for fusion to have gone past the power level 0
+	var/fusion_started = FALSE
+
+	///Stores the informations of the interface machine
+	var/obj/machinery/hypertorus/interface/linked_interface
+	///Stores the information of the moderator input
+	var/obj/machinery/atmospherics/components/unary/hypertorus/moderator_input/linked_moderator
+	///Stores the information of the fuel input
+	var/obj/machinery/atmospherics/components/unary/hypertorus/fuel_input/linked_input
+	///Stores the information of the waste output
+	var/obj/machinery/atmospherics/components/unary/hypertorus/waste_output/linked_output
+	///Stores the information of the corners of the machine
+	var/list/corners = list()
+	///Stores the information of the fusion gasmix
+	var/datum/gas_mixture/internal_fusion
+	///Stores the information of the output gasmix (used to move gases around, may be removed)
+	var/datum/gas_mixture/internal_output
+	///Stores the information of the moderators gasmix
+	var/datum/gas_mixture/moderator_internal
+	///Set the filtering type of the waste remove
+	var/filter_type = null
```

Comment:
When one of these objects gets deleted, does it clean up in here?

Reply:
most of them get cleaned up on the deletion of the core, if a machine gets deleted (should be able to be done only by admin) the check_part_connectivity() will null the one missing (but i suppose i could use signals, but i'm unsure how to do it in this case

Reply:
ok i used signals and they should all be deleted if one of the other objs get removed